pylgbst/utilities.py

```diff
@@ -14,18 +15,24 @@
 
 queue = queue  # just to use it
 
+def check_unpack(seq, index, pattern, size):
+    """Check that we got size bytes, if so, unpack using pattern"""
+    data = seq[index:index+size]
+    if len(data) == size:
```

Comment:
BTW did you get to understand why this can ever happen? I never saw messages to violate format expectations...

Reply:
No, it appeared on our very first run of the demo, so I was still working on the assumption that we'd messed something up in our setup and didn't investigate further. I observed that the code wasn't catching an error condition that obviously did occur and patched it when I looped back.